### DISCUSSION TITLE
Cannot paste images in MacOS screenshots (copied to clipboard)

### DISCUSSION THREAD

**Discussion by rnoxy** [xournalpp/xournalpp#4274] [Q&A]
Mac OS 12.6
Xournal v1.1.1 (stable)

Cannot paste the clipboard content.
I think this app need some permissions to access clipboard in MacOS.
Unfortunatelly I cannot find any seetings for this app.

Similarly, the app cannot acces files in "~/Documents".

**Comment by rolandlo** [comment]
The problem is that the Xournal++.app is not signed. Unfortunately code signing would be connected to a yearly fee paid to Apple, see https://github.com/xournalpp/xournalpp/pull/4103#issuecomment-1166308815. It's possible to self sign the app after which the permissions (like for accessing `"~/Documents"` are granted.

**Comment by flavinsh** [comment]
> Mac OS 12.6 Xournal v1.1.1 (stable)
> 
> Cannot paste the clipboard content. I think this app need some permissions to access clipboard in MacOS. Unfortunatelly I cannot find any seetings for this app.
> 
> Similarly, the app cannot acces files in "~/Documents".

Had the exact same issue: it was impossible to paste an image in Xournal++ (1.1.3) on macOS (12.6.5).
**Solution**: System Preferences > Security & Privacy > Privacy > Screen Recording > + > Xournal++

Seems to be working like a charm now.
